content/posts/gsoc-2024-auto-sync-sail/index.md

```diff
@@ -0,0 +1,151 @@
+Hello, I’m Mostafa. I graduated with Excellence from Cairo University’s Faculty of Engineering, Computer Engineering Department, class of 2023. I write C++ for a living. I love systems programming, metaprogramming & DSLs, as well as Compilers & VMs. You can find me @[Github](https://github.com/moste00), and @[Linkedin](www.linkedin.com/in/mostafa-m-kamal-b015691b1).
+
+I was honored to join GSoC in 2024, for my second (and last!) time, @Rizin org. The project that I applied for was the binary lifting of RISC-V instructions to Rizin’s bespoke internal representation, RzIL. However, updating the RISC-V capstone disassembler (originally a small task in the project preamble) turned out to need much more work than expected, and blocked the rest of the project.
+
+Let's start at the beginning.
+
+# RISC-V… Lifting?
+***Lifting*** is a term of art in compiler research and implementation, it refers to any process that takes as input a low-level machine code program and outputs a higher-level program. The reverse process, ***Lowering***, is what compilers do when they compile from a relatively high level language like C or LLVM IR to machine code. So you could simply think of Lifting as a synonym for “Decompiling” or “Reverse-Compiling”. 
+
+In the context of Rizin, Lifting refers to transforming a machine code program written for any of the hardware architectures that Rizin understands (x86, RISC-V, 6502, etc...) to a Rizin-specific intermediate language called RzIL. 
+
+By doing this, Rizin’s developers can write generic analysis algorithms that manipulate RzIL instructions, and a generic VM that executes Rizin instructions, only once. Then, for each architecture that Rizin supports, a lifter that transforms machine code written for that architecture into RzIL is written, and we get all the analysis algorithms and VM execution capabilities “for free”. 
+
+In a nutshell, RzIL is the universal “Lingua Franca” for Rizin, like English is for Software Engineering.
+
+
+![IMAGE__WORLD_WITHOUT_RZIL](https://hackmd.io/_uploads/HyfLVzMHkg.png)
+Figure 1: Without RzIL, there is no smarter way to perform N operations for M assembly languages other than doing an NxM amount of work, implementing the N operations over and over again per each language/architecture.
+
+![IMAGE__WORLD_WITH_RZIL](https://hackmd.io/_uploads/HklxSGfHyg.png)
```

Comment:
please push all the pngs in the same folder as the index.md

then fix the links like in the example below: https://github.com/rizinorg/website/blob/fc869385e850146f02cc36697b38f27402f2b6ce/content/posts/gsoc-2021-summary/index.md?plain=1#L120